redash/query_runner/mongodb.py

```diff
@@ -102,14 +102,55 @@ def _get_column_by_name(self, columns, column_name):
 
         return None
 
-
-    def run_query(self, query):
+    def _get_db(self):
         if self.is_replica_set:
             db_connection = pymongo.MongoReplicaSetClient(self.configuration["connectionString"], replicaSet=self.configuration["replicaSetName"])
         else:
             db_connection = pymongo.MongoClient(self.configuration["connectionString"])
 
-        db = db_connection[self.db_name]
+        return db_connection[self.db_name]
+
+    def _merge_property_names(self, columns, document):
+        for property in document:
+              if property not in columns:
+                  columns.append(property)
```

Comment:
```python
return [prop for prop in document if prop not in columns]`
```

but you can use `set`s here (which will also be more performant, although it's probably negligible): 
```python
set(columns) | set(document.keys()) # or columns can be a set from the get-go.
```